internal/controller/ws/client.go

```diff
@@ -27,6 +26,8 @@ type rcClient struct {
 
 	responseDispatcher *controller.ResponseDispatcher
 
+	messageDispatcher *controller.MessageDispatcher
```

Comment:
Just thinking a bit here...  In this case, I don't think the client/peer needs to keep up with the message dispatcher object.  You don't have to make any changes in this PR.  If we think its not really needed, we can remove it another PR.

Reply:
Will add in a follow up PR